### ISSUE TITLE
Modulariação das extensões

### ISSUE THREAD

**Issue by prhost** [prhost/new-opencart#2]
Aqui meus amigos, esta um dos maiores diferenciais do "novo opencart". Criar as extensões em padrões modulares é a solução de muitos problemas, assim evitamos misturas de arquivos, conflitos e modificacoes no core, facilitando o gerenciamento da instalação e remoção de uma ou mais extensões, e entre outros benefícios.

![oc - extensoes](https://cloud.githubusercontent.com/assets/3984760/12620071/e67527f6-c501-11e5-9816-8a0c725fcfde.jpg)

**Comment by felipoantonoff** [comment]
Excelente, acho que fica mais organizado e cada extenção tem sua pasta, sendo que para apagar é só remover ela.
Mudando a estrutura, também é bom modificar o Instalador nativo para seguir o novo padrão, se bem que pelo que vi, ele segue o padrão que adicionamos dentro do upload, não precisando mexer. 
Agora esse vQmod para cada modificação, seria apenas o xml ou o vQmod mesmo?

**Comment by prhost** [comment]
@FelipoCodeMarket é só o xml mesmo, esqueci de abrir a estrutura dele mas la dentro tem apenas `vqmod/xml/arquivo.xml` já o instalador nativo na minha opinião, não gosto muito dele, acho que é desnecessário, se usarmos o composer certinho no final das contas basta apenas vc executar o comando `composer create-project new-opencart` e depois configurar o arquivo `configs/database.php` ta pronto sua loja (:

**Comment by prhost** [comment]
Outra coisa que precisaria ser forte levando em conta a segurança é o htaccess e o ngnx, eu trabalho com apache e fiz as modificações necessárias por mais que eu sempre vou achar que htaccess é uma parada alienígena enviada para nos aniquilar, já ngnx não manjo, teria que ter alguem da equipe que entenda e transforme o htaccess em ngnx. Falo isso para proteger aruqivos como vendor, script sql, logs, etc.

**Comment by joelburigo** [comment]
Eu já tive algum contato com nginx, todos os Opencart que estou instalando agora só nginx, acho baca oferecer suporte ao php7 também.

**Comment by prhost** [comment]
@joelmartins é então, nunca testei rodar, temos que ver quais são as incompatibilidade e tentar ajustar, mas sim não vamos nos prender a versões antigas do php, uma vez que podemos da suporte ao 7 é um grande avanço.

**Comment by arnisjuraga** [comment]
Hi guys, 

i have worked with opencart and vqmod for a while, but I really believe - modulation system with " search-replace"  in source code is something, we can get rid off at all :(

The only drawback of that is that it will not support any modules or themes developed for that.
Although, you can keep support for vqmod, but as more changes you will make in original source, as less modules will be supported, because the relay on source structure and source "text".

So, this is quite serious decision to make.  And in this point - I am convicted - we need to develop modularisation engine based on Opencart events and triggers, or something completely new.

**Comment by prhost** [comment]
@arnisjuraga to remove full vqmod and ocmod, need to let modulate all posible partial have change, dashboard, menu, filters, settings, etc, which is admin. The theme unfortuntely has to be replace the entire view and not part of it.

**Comment by arnisjuraga** [comment]
Yes, this is the catch - we will keep VQMOD support, thats not a problem.

Problem is - that as soon as we start to change original core code, most of the plugins will automatically become incompatible. As VQMOD and OCMOD works just with search/replace. 

And that means - as soon we dcide to "drop" it, we need to think about alternatives - how to create pluggable environment for developers.
